### PR TITLE
RSDK-3162 implement proper closeout and mutex for rplidar

### PR DESCRIPTION
**Pull request by kim-mishra**
[Ticket](https://viam.atlassian.net/browse/RSDK-3162)

## Changes
- adding lock around close logic
- adding closed bool to struct 
- throw closed error if we try to call `NextPointCloud` while closed

## Manual testing
- rplidar module works before and after changes

### CODE REVIEW

File: rplidar.go [hunk 1]
```diff
@@ -210,6 +216,10 @@ func (rp *Rplidar) Stream(ctx context.Context, errHandlers ...gostream.ErrorHand
```
Comment (kim-mishra) [inline rplidar.go:216]: Do we want a resource is closed error for these functions or is unimplemented sufficient?

**Comment by EshaMaharishi** [comment]
(removing myself from the review)

File: rplidar.go (hunk 1 above)
Comment (nicksanford) [reply rplidar.go:216]: unimplemented is sufficient

**Review by nicksanford — APPROVED** [review]
This looks great. Normally I would ask for you to add tests but given that there are currently no tests for RPlidar I've added a ticket to create tests in the future: https://viam.atlassian.net/browse/RSDK-3523

**Pull request merged**